annotation_pipeline/populate_labelstudio.py

```diff
@@ -0,0 +1,172 @@
+    """
+    Prompts terminal to run tag collector on crawled URL's
+    filename: name of csv containing crawled URL's
+
+    CSV of URL's + collected tags saved in ./labeled-source-text.csv
+    """
+    tag_collector = f"python3 ../html_tag_collector/collector.py common_crawler/{filename} --render-javascript"
+
+    return_code, stdout, stderr = run_subprocess(tag_collector)
+
+    return return_code, stdout, stderr
+
+def csv_to_label_studio_tasks(csv_file_path: str, batch_id: str, output_name: str) -> list[dict]:
+    """
+    Formats CSV into list[dict] with "data" key as labelstudio expects
+    csv_file_path: path to csv with labeled source text
+    batch_id: timestamp to append to all URL's in batch
+    output_name: saves tag_collected CSV + batch_info in tag_collector/{output_name}
+    """
+    df = pd.read_csv(csv_file_path)
+    df['batch_id'] = [batch_id] * len(df)
+    df = df.fillna('')
+    df.to_csv("tag_collector/" + output_name.replace("urls/", "", 1), index=False)
+    tasks = [{"data": row.to_dict()} for _, row in df.iterrows()]
+    return tasks
+
+def get_huggingface_repo_id(config_file: str) -> str:
+    config = configparser.ConfigParser()
+    config.read(config_file)
+
+    # Retrieve the huggingface_repo_id from the DEFAULT section
+    huggingface_repo_id = config['DEFAULT'].get('huggingface_repo_id')
+    
+    if huggingface_repo_id is None:
+        raise ValueError("huggingface_repo_id not found in the config file.")
+
+    return huggingface_repo_id
+
+def main():
+    """
+    This script automates the process of crawling for relevant URL's,
+    scraping HTML content from those pages, formatting the data as label studio tasks,
+    and uploading to label studio
+    """
+
+# ----------------- COMMON CRAWL -----------------------------------
+
+    #common crawl parameters
+    common_crawl_id = 'CC-MAIN-2024-10'
+    url = '*.gov'
+    search_term = 'police'
+    num_pages = '2'
+
+    #run common crawl
+    crawl_return_code, crawl_stdout, crawl_stderr = run_common_crawl(common_crawl_id, url, search_term, num_pages)
+
+    #check success
+    if crawl_return_code != 0:
+        print(f"Common crawl script failed:\n{crawl_stderr}")
+        return
+
+    #print batch info to verify before continuing
+    batch_info = pd.read_csv("common_crawler/data/batch_info.csv").iloc[-1]
+    print("Batch Info:\n" + f"{batch_info}")
+
+    if(batch_info["Count"] != 0):
+        input("Confirm batch info and csv on HuggingFace, press any key to proceed. Ctrl-c to quit")
+    else:
+        return
+
+    #get urls from hugging face
+    #REPO_ID = "PDAP/unlabeled-urls"
+    REPO_ID = get_hugging_face_repo_id("../common_crawler/config.ini")
+    FILENAME = "urls/" + batch_info["Filename"] + ".csv"
+    df = pd.read_csv(hf_hub_download(repo_id=REPO_ID, filename=FILENAME, repo_type="dataset", local_dir="common_crawler/"), index_col=False)
+
+# ----------------- TAG COLLECTOR ----------------------------------
```

Comment:
NIT: As a general rule, if you're breaking down code into a commented section, you can probably extract that section to a function to increase readability. Recommend doing that here, but won't block on it.